go/vt/vtgate/engine/limit.go

```diff
@@ -108,6 +110,8 @@ func (l *Limit) TryStreamExecute(ctx context.Context, vcursor VCursor, bindVars
 			return nil
 		}
 
+		mu.Lock()
+		defer mu.Unlock()
```

Comment:
Looks like need this for the count, but wonder, should we do this with an atomic counter instead? The usage might be complicated though, so the lock might be a lot easier for now and then to optimize later.